flax/experimental/nnx/nnx/graph_utils.py

```diff
@@ -523,16 +595,25 @@ def _get_children():
   if isinstance(node_impl, MutableNodeImpl):
     # we create an empty node first and add it to the index
     # this avoids infinite recursion when there is a reference cycle
-    node = node_impl.create_empty(graphdef.metadata)
-    index_to_node[graphdef.index] = node
+    if ref_cache is not None and graphdef.index in ref_cache:
+      node = ref_cache[graphdef.index]
+      if type(node) != graphdef.type:
+        raise ValueError(
+          f'Expected a node of type {graphdef.type.__name__} for index '
```

Comment:
Is there a reason for printing type names here and type objects elsewhere (e.g. on L579 above)?

Reply:
not sure, reverted it back to the full type